File: app/services/group_service.py
import random
import string
from app.core.db import db
from fastapi import HTTPException

def generate_invite_code(length=6):
    """6 haneli, büyük harf ve rakamlardan oluşan rastgele kod üretir."""
    chars = string.ascii_uppercase + string.digits
    return ''.join(random.choices(chars, k=length))

async def create_group(name: str):
    # Benzersiz bir davet kodu üretilene kadar dene
    while True:
        code = generate_invite_code()
        existing = await db.group.find_unique(where={"inviteCode": code})
        if not existing:
            break
    
    # Grubu veritabanına kaydet
    new_group = await db.group.create(
        data={
            "name": name,
            "inviteCode": code
        }
    )
    return new_group

async def join_group(invite_code: str, user_id: int):
    # Grubu davet koduyla bul
    group = await db.group.find_unique(where={"inviteCode": invite_code})
    
    if not group:
        # Geçersiz kod hatası (404)
        raise HTTPException(status_code=404, detail="Geçersiz Kod - Grup bulunamadı.")
    
    # Kullanıcıyı güncelle ve gruba dahil et
    updated_user = await db.user.update(
        where={"id": user_id},
        data={"groupId": group.id}
    )
    
    return group

File: src/services/group.service.ts
import prisma from '../config/db';

/**
 * Kuruşla Paylaş — grup yönetimi
 */
export class GroupService {
  static async createGroup(name: string, creatorUserId?: number) {
    const inviteCode = Math.random().toString(36).substring(2, 8).toUpperCase();

    const group = await prisma.group.create({
      data: { name, inviteCode },
    });

    if (creatorUserId) {
      await prisma.user.update({
        where: { id: creatorUserId },
        data: { groupId: group.id },
      });
    }

    return group;
  }

  /**
   * Davet kodu ile gruba katılır (Kuruşla Paylaş)
   */
  static async joinGroup(inviteCode: string, userId: number) {
    const group = await prisma.group.findUnique({
      where: { inviteCode: inviteCode.toUpperCase() },
    });

    if (!group) {
      throw new Error('Geçersiz davet kodu.');
    }

    const user = await prisma.user.findUnique({ where: { id: userId } });
    if (!user) {
      throw new Error('Kullanıcı bulunamadı.');
    }

    if (user.groupId && user.groupId !== group.id) {
      throw new Error('Kullanıcı zaten başka bir grupta.');
    }

    return prisma.user.update({
      where: { id: userId },
      data: { groupId: group.id },
      include: {
        group: {
          include: {
            pots: { where: { isActive: true }, include: { participants: true } },
            users: { select: { id: true, email: true, firstName: true, lastName: true } },
          },
        },
      },
    });
  }

  static async getGroupById(groupId: number) {
    return prisma.group.findUnique({
      where: { id: groupId },
      include: {
        users: { select: { id: true, email: true, firstName: true, lastName: true } },
        pots: {
          where: { isActive: true },
          include: {
            participants: {
              include: {
                user: { select: { id: true, email: true, firstName: true, lastName: true } },
              },
            },
          },
        },
      },
    });
  }
}
